server/api/src/rsGeneralAdmin.cpp

```diff
@@ -1135,9 +1139,43 @@ _rsGeneralAdmin( rsComm_t *rsComm, generalAdminInp_t *generalAdminInp ) {
     }
     if ( strcmp( generalAdminInp->arg0, "rm" ) == 0 ) {
         if ( strcmp( generalAdminInp->arg1, "user" ) == 0 ) {
-            return irods::remove_user(*rsComm,
-                                      generalAdminInp->arg2 ? generalAdminInp->arg2 : "",
-                                      generalAdminInp->arg3 ? generalAdminInp->arg3 : "");
+            const auto* user_name = generalAdminInp->arg2 ? generalAdminInp->arg2 : "";
+            const auto* zone_name = generalAdminInp->arg3 ? generalAdminInp->arg3 : "";
+            namespace iua = irods::experimental::administration;
+            const auto user_and_zone = irods::user::validate_name(user_name);
+            // new functionality only enforced on versions >= 4.3.2
+            const auto client_version = irods::to_version(rsComm->cliVersion.relVersion);
+            constexpr irods::version v4320{4, 3, 2};
+            if (user_and_zone != std::nullopt && client_version != std::nullopt && client_version >= v4320) {
```

Comment:
Avoid implicit conversions when interacting with the contents of an optional.

Instead of using `client_version >= v4320`, prefer one of the following:
- `*client_version >= v4320`
- `client_version.value() >= v4320`